### ISSUE TITLE
Should this be merged to Go core?

### ISSUE THREAD

**Issue by nikolaydubina** [alphadose/ZenQ#1]
If this is really fast and without downsides, should this be in Go core?

Maybe folks in go-dev google group will get interest.

**Comment by shubhang93** [comment]
I don’t think so, although it looks fast on paper, one of the biggest advantages of go’s native channel is the ability to select across multiple channels and broadcast using the close function

**Comment by nikolaydubina** [comment]
Select, Close, Broadcast sounds like additional features, possibly implemented without deteriorating performance boost?

I am very curious what are the drawbacks of this approach. If there are no clear problems, this can be big thing like [Ryu](https://github.com/golang/go/issues/15672) or  more recent [pdqsort](https://github.com/golang/go/commit/72e77a7f41bbf45d466119444307fd3ae996e257). 

I think we are standing on something bi here.

**Comment by nikolaydubina** [comment]
cc @alphadose

**Comment by alphadose** [comment]
I have already opened an issue for it https://github.com/golang/go/issues/52652

**Comment by alphadose** [comment]
We can gain even further optimizations by using runtime internal methods like gopark instead of spinning mechanism which is the current implementation

**Comment by MeteorsLiu** [comment]
Hope Go can support this optimization. It's a certain great project!
However, as far as i'm concerned, those guys in Google wonld't hear from people from the community most of time. That's a little bit tragedy. :(
